### PR TITLE
Prepare for v2.2.2 release

### PR DESCRIPTION
**Pull request by Jdubrick**
### What does this PR do?:

This PR prepares for the v2.2.2 release to be cut. It updates the api schema to v2.2.2 and updates the reference to the schema documentation to properly point to v2.2.2. Additionally, when updating the api schema to v2.2.2 some tests were failing, this is because of changes made in the api [here](https://github.com/devfile/api/pull/1217) that included an extra field to variable warnings.

### Which issue(s) this PR fixes:

Fixes https://github.com/devfile/api/issues/1477
### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened and linked to this PR, if they are not in the PR scope due to various constraints.

- [ ] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [ ] Documentation (READMEs, Product Docs, Blogs, Education Modules, etc.)

   

- [ ] Client Impact

  

- [ ] Gosec scans

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/devfile/library/v2
 go 1.19
 
 require (
-	github.com/devfile/api/v2 v2.2.1
+	github.com/devfile/api/v2 v2.2.2
 	github.com/devfile/registry-support/registry-library v0.0.0-20240311160550-e51ee8934746
```
Comment (michael-valdron) [inline go.mod:7]: There was some additional changes to registry-library that should be bumped to include in this devfile/library release:

```suggestion
	github.com/devfile/registry-support/registry-library v0.0.0-20240328155806-7c89891a72ce
```

**Review by michael-valdron — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (michael-valdron) [reply go.mod:7]: These two commit specifically:
- https://github.com/devfile/registry-support/commit/9445a839c89f7a12336161c018a6d020b983840c
- https://github.com/devfile/registry-support/commit/c12332d7588042539f49e48fdfccdc877f2f332a

Comment (Jdubrick) [reply go.mod]: Good catch thanks :)

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Pull request merged**